data_api3/reader.py

```diff
@@ -194,7 +194,7 @@ def process_channel_header(msg):
 
     if compression == 0:
         # NOTE legacy compatibility: historically a shape [1] is treated as scalar
-        if shape is None or shape == [1]:
+        if shape is None or len(shape) == 0 or shape == [1]:
```

Comment:
Just as a side note, [1] comes from the Epics world, actually there everything is an "array".